python/cugraph/cugraph/structure/graph_implementation/simpleGraph.py

```diff
@@ -152,12 +152,20 @@ def __from_edgelist(
             # The symmetrize step may add additional edges with unknown
             # ids and types for an undirected graph.  Therefore, only
             # directed graphs may be used with ids and types.
-            if len(edge_attr) == 3 and not self.properties.directed:
-                raise ValueError(
-                    "User-provided edge ids and edge "
-                    "types are not permitted for an "
-                    "undirected graph."
-                )
+            if len(edge_attr) == 3:
+                if not self.properties.directed:
+                    raise ValueError(
+                        "User-provided edge ids and edge "
+                        "types are not permitted for an "
+                        "undirected graph."
+                    )
+                if not legacy_renum_only:
+                    raise ValueError(
+                        "User-provided edge ids and edge "
+                        "types are only permitted when "
+                        "from_edgelist is called with "
+                        "legacy_renum_only=True."
```

Comment:
Just ran into this. Why do we need this again ? I am running into some memory issues with `legacy_renum_only ` on SG Graphs on this PR

Reply:
The non-legacy renumbering removes the additional edge properties.